sunpy/coordinates/transformations.py

```diff
@@ -118,15 +118,28 @@ def _observers_are_equal(obs_1, obs_2, string_ok=False):
     if string_ok:
         if obs_1 == obs_2:
             return True
-    if not (isinstance(obs_1, BaseCoordinateFrame) and isinstance(obs_2, BaseCoordinateFrame)):
-        raise ValueError("To compare two observers, both must be instances of BaseCoordinateFrame. "
-                         "Cannot compare two observers {} and {}.".format(obs_1, obs_2))
+
+    if not isinstance(obs_1, BaseCoordinateFrame):
+        raise ConvertError(f"Source observer {obs_1} is not fully specified.")
```

Comment:
I think the original message was clearer here, ie. mentioning explicitly that `obs_1` must be an instance of `BaseCoordinateFrame`.

Reply:
I've made the error messages more helpful